desktop/src/com/frostwire/gui/bittorrent/CreateTorrentDialog.java

```diff
@@ -286,6 +321,21 @@ public void mouseClicked(MouseEvent e) {
                 }
             }
         });
+
+        pieceSizeComboBox.addActionListener(
+                new ActionListener() {
+                    @Override
+                    public void actionPerformed(ActionEvent e) {
+                        onPieceSizeSelected(pieceSizeComboBox.getSelectedIndex());
+                    }
+                }
+        );
+    }
+
+    private void onPieceSizeSelected(int selectedIndex) {
+        if (selectedIndex > 0 && values.length > selectedIndex && values[selectedIndex] != null) {
+            pieceSize = values[selectedIndex].kb*_1024BYTES;
```

Comment:
and you could make the `.kb` property `private` inside the enum.
In the end, the user only needs to see the human representation for the combobox.